core/src/test/java/org/kiwix/kiwixmobile/core/page/history/viewmodel/HistoryViewModelTest.kt

```diff
@@ -57,6 +57,7 @@ internal class HistoryViewModelTest {
     every { zimReaderContainer.name } returns "zimName"
```

Comment:
I would expect to see a lot of deletion in `Bookmarks/HistoryViewmodelTest` because a test should only tests what is in a class and you have moved a lot to the parent. I would want to see `PageViewModelTest` that works through the common codepaths, to facilitate this you will have to create a testable version of this abstract class, I would have it in `PageViewModelTest` file and I generally go with `TestablePageViewModel`

Reply:
@Frans-Lukas this is the only large piece left I think, also your unit tests are failing, mostly over usage of `filtered` instead of `visible` items I would imagine. Similarly with how I want the abstract classes tested and the implementations to only test the code contained in the class I would expect to see `filteredItems` tested in `PageStateTest` and nowhere else

Reply:
Same as my comment on BookmarkViewModelTest

Reply:
@Frans-Lukas I think you didn't see this comment because I appended it on to this thread